pkg/agent/run.go

```diff
@@ -138,14 +150,35 @@ func Run(cmd *cobra.Command, args []string) {
 
 	for {
 		log.Println("Running Agent...")
-		log.Println("Posting data to ", serverURL)
-		err = preflightClient.PostDataReadings(readings)
-		// TODO: handle errors gracefully: e.g. handle retries when it is possible
-		if err != nil {
-			log.Fatalf("Post to server failed: %+v", err)
+		log.Println("Posting data to ", baseURL)
+		if config.OrganizationID == "" {
+			data, err := json.Marshal(readings)
+			if err != nil {
+				log.Fatalf("Cannot marshal readings: %+v", err)
+			}
+			path := config.Endpoint.Path
+			if path == "" {
+				path = "/api/v1/datareadings"
+			}
+			res, err := preflightClient.Post(path, bytes.NewBuffer(data))
+			if code := res.StatusCode; code < 200 || code >= 300 {
```

Comment:
Should 3xx responses not be ok?

Reply:
I guess the client will follow redirects but if we get another 300 response here we don't know what to do so what you have is good I think.

Reply:
I don't think so.

Maybe we can make the client handle 3xx responses. The common cases (as the redirect) are already handled by the go `http` client by default.